src/engine.c

```diff
@@ -4548,6 +4733,50 @@ int settle_callback(game *g, int who, int which, int list[], int num,
 				if (!g->simulation)
 				{
 					/* Format message */
+					sprintf(msg, "%s discards a Rare good for "
+					        "extra military.\n", p_ptr->name);
+
+					/* Send message */
+					message_add(g, msg);
+				}
+
+				/* Need one more rare good */
+				goods_needed[GOOD_RARE]++;
+			}
+
+			/* Check for consume to increase military */
+			if (o_ptr->code & P3_CONSUME_ALIEN)
+			{
+				/* Mark power as used */
+				c_ptr->misc |= 1 << (MISC_USED_SHIFT + j);
+
+				/* Ask for goods to consume later */
+				consume_military++;
+
+				if (o_ptr->code & P3_XENO)
+				{
+					if (t_ptr->d_ptr->flags & FLAG_XENO)
+					{
+						/* Add extra military */
+						military += o_ptr->value;
+					}
+
+					/* Remember bonus for later */
+					p_ptr->bonus_military_xeno += o_ptr->value;
+				}
+				else
+				{
+					/* Add extra military */
+					military += o_ptr->value;
+
+					/* Remember bonus for later */
+					p_ptr->bonus_military += o_ptr->value;
+				}
+
+				/* Message */
+				if (!g->simulation)
+				{
+					/* Format message */
```

Comment:
This message needs to be updated to say "for extra military against Xenos" (when applicable). Perhaps it should be put inside the respective branches above.